### DISCUSSION TITLE
Wrong legend is added

### DISCUSSION THREAD

**Discussion by zhengliuX** [pyvista/pyvista#1933] [Q&A]
Hi, I was trying to build an app, with Pyvistaqt. It involves some interactivate plot, like add a mesh/actor, or remove it, or add it again.
I now encountered one problem.
I add two mesh, set as two actors, then I remove themm all (also remove_legend()), add one of them again(also add_legend() again), then the legend did not show right.
Here is the code:
```

import pyvista as pv

plotter = pv.Plotter()

cube1 = pv.Cube([0, 0, 0], 1, 3, 6)
cube1_actor = plotter.add_mesh(cube1, color='r', label='red cube')

cube2 = pv.Cube([3, 3, 3], 1, 3, 6)
cube2_actor = plotter.add_mesh(cube2, color='b', label='blue cube')

plotter.add_legend()

plotter.remove_actor(cube1_actor)
plotter.remove_actor(cube2_actor)
plotter.remove_legend()

plotter.add_actor(cube1_actor)
plotter.add_legend()

plotter.camera_position = 'yz'
plotter.show()
```
![image](https://user-images.githubusercontent.com/65115609/145047081-a7088016-7118-45b4-9685-ca1e471ba079.png)
You can see that I remove all the actors, then add cube1_actor again, but it shows me all the legend

**Comment by banesullivan** [comment]
This indeed looks like a bug to me. I'm going to convert it to an issue

**Reply by banesullivan** [reply]
Please follow up in #1934 - I'm going to lock this discussion to keep everything in one place
